modules/apps/29-fee/ibc_module.go

```diff
@@ -187,7 +198,23 @@ func (im IBCModule) OnAcknowledgementPacket(
 	acknowledgement []byte,
 	relayer sdk.AccAddress,
 ) error {
-	// TODO: Implement fee specific logic if fee is enabled for the given channel
+	if im.keeper.IsFeeEnabled(ctx, packet.SourcePort, packet.SourceChannel) {
+		ack := &types.IncentivizedAcknowledgement{}
+		if err := types.ModuleCdc.UnmarshalJSON(acknowledgement, ack); err != nil {
+			return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-29 incentivized packet acknowledgement: %v", err)
+		}
+
+		packetId := channeltypes.NewPacketId(packet.SourceChannel, packet.SourcePort, packet.Sequence)
+		identifiedPacketFee, found := im.keeper.GetFeeInEscrow(ctx, packetId)
+		if !found {
+			return sdkerrors.Wrapf(types.ErrFeeNotFound, "fee not found for packet id %s", packetId)
```

Comment:
Don't return error here, since it's possible to send a packet over a fee-enabled channel without escrowing fees.

Just move on to the underlying app